src/nvim/memline.c

```diff
@@ -1767,7 +1767,7 @@ ml_get_buf (
    * Don't use the last used line when 'swapfile' is reset, need to load all
    * blocks.
    */
-  if (buf->b_ml.ml_line_lnum != lnum || mf_dont_release) {
+  if (buf->b_ml.ml_line_lnum != lnum) {
```

Comment:
Perhaps a problem with missing `mf_dont_release == true` flag if we come from [here](https://github.com/neovim/neovim/pull/4486/files#diff-267fc841fb61a510c195c6d3b37b4b06R200) and want blocks in memory?

If `lnum == 1` and the last cached line is also `1` then a "dirty" line would not be flushed/block loaded. Or is that not a problem because the old block would still be the same (and in memory)?